code/game/mecha/equipment/tools/tools.dm

```diff
@@ -1296,6 +1307,8 @@
 /obj/item/mecha_parts/mecha_equipment/tool/switchtool/Destroy()
 	qdel(switchtool)
 	switchtool = null
+	qdel(pr_switchtool)
```

Comment:
Does this actually cause the iterators to get shut down correctly, though? Just checking since mech code is ancient.

Reply:
I'm not sure how I would check for that. I think global iterators have a `stop()` function, should I stop them before deleting them?

Reply:
No I meant you should check the global iterator code to see if they handle `Destroy()` at all.

I just checked, they definitely don't, so add destruction code to clean them up completely. https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/datums/helper_datums/global_iterator.dm

Reply:
I've added a `Destroy()` to global iterators.